Source/DafnyTestGeneration/Main.cs

```diff
@@ -110,9 +110,9 @@ public static class Main {
         yield break;
       }
       var dafnyInfo = new DafnyInfo(program);
-      var rawName = sourceFile.Split("/").Last().Split(".").First();
+      var rawName = sourceFile.Split(new[] { '/', '\\' }).Last().Split(".").First();
```

Comment:
I would definitely agree on this readability refactoring on that but it would need to be tested, since they don't have exactly the same semantics.
`GetFileNameWithoutExtension` returns the filename minus the last period and everything that follows.
This code returns the filename minus the first period and everything that follows.
So for example, in the case of a file `file.dfy.expect`, this code would return `file`, but `Path.GetFileNameWithoutExtension` would return `file.dfy`.
I'm not sure it's a problem though, but I would have to run the full CI on that to ensure it's working so let's keep it for the very last test, for when there will be no CI test failing (currently there are a lot).

Reply:
So, now then?

Reply:
I'm ready to answer this now. Looking at the code more carefully, this `rawName` should not contain any dot since it is used as a module name in a dafny tile. However, the function `Path.GetFileNameWithoutExtension` does not guarantee that the final name won't have a dot. But the construct above does guarantee the absence of dot in the final name.
Note that I also found some Dafny files that look like `filename.dfy.tmp.dprint.dfy` and that would thus crash the test, so even though this would not be a problem now, it's better to not introduce potential bugs.
Thus, let's keep it the way it is (at least for the dot part).

Reply:
I still refactored the code the following way:

`Path.GetFileName(sourceFile).Split(".").First()`

Reply:
I'm confused. Don't we want it to assign `filename.dfy.tmp.dprint` in case the sourceFile is `filename.dfy.tmp.dprint.dfy` ?

Reply:
I frankly don't know what exactly we want here, because I was just fixing the filename extraction so that it works on Windows. The documentation about this method is here and dates from earlier in March:
https://github.com/Dargones/dafny/tree/PatchesToCounterExamples/Source/DafnyTestGeneration#readme
As you can read, the name generation is under-specified because it assumes a file name with exactly one dot.

If you have a clear vision (I don't) and are sure of the intended behavior, then I'll do the change because it's cheap. I just checked that Dafny accepts dots in module names, so lights are green on that. Otherwise; I'd suggest we leave it for when it will be needed.

Reply:
Let's leave it since it seems out of scope for this PR :-)